### PR TITLE
Version 2.1.0

### PR DESCRIPTION
**Pull request by yafuquen**
## Description
- Improve input errors
- Support notification extensions

## Testing
- [x] Added unit tests
- [x] Ran unit tests successfully
- [x] Added documentation for public APIs and/or Wiki

### CODE REVIEW

**Review by mejiagarcia — APPROVED** [review]

File: TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift [hunk 1]
```diff
@@ -29,32 +29,58 @@ class PushChallengeProcessor {
...
+    failure: @escaping TwilioVerifyErrorBlock
+  ) {
     Logger.shared.log(withLevel: .info, message: "Getting challenge \(sid) with factor \(factor.sid)")
     challengeProvider.get(withSid: sid, withFactor: factor, success: success, failure: { error in
       failure(TwilioVerifyError.inputError(error: error as NSError))
     })
   }
   
-  func updateChallenge(withSid sid: String, withFactor factor: PushFactor, status: ChallengeStatus, success: @escaping EmptySuccessBlock, failure: @escaping TwilioVerifyErrorBlock) {
+  func updateChallenge(
```
Comment (AuthyApps) [inline TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift:54]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/0307d478e851ddee7e698ae03678695d1c0f326c/TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift#L54"></span>Function should have complexity 10 or less: currently complexity equals 11<br />
<code>cyclomatic_complexity</code> <code>PushChallengeProcessor.swift:54</code></td>
    </tr>
  </tbody>
</table>

Comment (AuthyApps) [inline TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift:54]: <table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/0307d478e851ddee7e698ae03678695d1c0f326c/TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift#L54"></span>Function body should span 40 lines or less excluding comments and whitespace: currently spans 60 lines<br />
<code>function_body_length</code> <code>PushChallengeProcessor.swift:54</code></td>
    </tr>
  </tbody>
</table>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`93.61%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Key/Signer/ECSigner.swift | **`94.74%`**
TwilioVerifySDK/TwilioSecurity/Sources/Key/Template/ECP256SignerTemplate.swift | **`92.11%`**
TwilioVerifySDK/TwilioSecurity/Sources/Keychain/KeyManager.swift | **`85.71%`**
TwilioVerifySDK/TwilioSecurity/Sources/Keychain/Keychain.swift | **`92.45%`**
TwilioVerifySDK/TwilioSecurity/Sources/Keychain/KeychainQuery.swift | **`100.00%`**
TwilioVerifySDK/TwilioSecurity/Sources/Storage/SecureStorage.swift | **`84.62%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/JWT/JwtGenerator.swift | **`100.00%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/JWT/JwtSigner.swift | **`74.75%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/KeyStorage.swift | **`92.50%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/Storage+AccessGroup.swift | **`93.33%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/Storage.swift | **`90.84%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/StorageProvider.swift | **`80.85%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeFacade.swift | **`96.63%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift | **`94.12%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift | **`96.91%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`97.60%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`98.83%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`79.41%`**
TwilioVerifySDK/TwilioVerify/Sources/Networking/Authentication.swift | **`100.00%`**
TwilioVerifySDK/TwilioVerify/Sources/TwilioVerifyError.swift | **`94.32%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by SergioFierro — APPROVED** [review]

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`93.74%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Key/Signer/ECSigner.swift | **`94.74%`**
TwilioVerifySDK/TwilioSecurity/Sources/Key/Template/ECP256SignerTemplate.swift | **`92.11%`**
TwilioVerifySDK/TwilioSecurity/Sources/Keychain/KeyManager.swift | **`85.71%`**
TwilioVerifySDK/TwilioSecurity/Sources/Keychain/Keychain.swift | **`92.45%`**
TwilioVerifySDK/TwilioSecurity/Sources/Keychain/KeychainQuery.swift | **`100.00%`**
TwilioVerifySDK/TwilioSecurity/Sources/Storage/SecureStorage.swift | **`84.62%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/JWT/JwtGenerator.swift | **`100.00%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/JWT/JwtSigner.swift | **`74.75%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/KeyStorage.swift | **`92.50%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/Storage+AccessGroup.swift | **`93.33%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/Storage.swift | **`90.84%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/StorageProvider.swift | **`80.85%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeFacade.swift | **`96.63%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift | **`94.12%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift | **`96.91%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`97.60%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`98.83%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`78.12%`**
TwilioVerifySDK/TwilioVerify/Sources/Networking/Authentication.swift | **`100.00%`**
TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkAdapter.swift | **`97.83%`**
TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider+Models.swift | **`100.00%`**
TwilioVerifySDK/TwilioVerify/Sources/TwilioVerifyError.swift | **`94.32%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`93.74%`**
File | Coverage
-----|-----
TwilioVerifySDK/TwilioSecurity/Sources/Key/Signer/ECSigner.swift | **`94.74%`**
TwilioVerifySDK/TwilioSecurity/Sources/Key/Template/ECP256SignerTemplate.swift | **`92.11%`**
TwilioVerifySDK/TwilioSecurity/Sources/Keychain/KeyManager.swift | **`85.71%`**
TwilioVerifySDK/TwilioSecurity/Sources/Keychain/Keychain.swift | **`92.45%`**
TwilioVerifySDK/TwilioSecurity/Sources/Keychain/KeychainQuery.swift | **`100.00%`**
TwilioVerifySDK/TwilioSecurity/Sources/Storage/SecureStorage.swift | **`84.62%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/JWT/JwtGenerator.swift | **`100.00%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/JWT/JwtSigner.swift | **`74.75%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/KeyStorage.swift | **`92.50%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/Storage+AccessGroup.swift | **`93.33%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/Storage.swift | **`90.84%`**
TwilioVerifySDK/TwilioVerify/Sources/Data/StorageProvider.swift | **`80.85%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeFacade.swift | **`96.63%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/ChallengeRepository.swift | **`94.12%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Challenge/PushChallengeProcessor.swift | **`96.91%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`97.60%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`98.83%`**
TwilioVerifySDK/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`78.12%`**
TwilioVerifySDK/TwilioVerify/Sources/Networking/Authentication.swift | **`100.00%`**
TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkAdapter.swift | **`97.83%`**
TwilioVerifySDK/TwilioVerify/Sources/Networking/NetworkProvider+Models.swift | **`100.00%`**
TwilioVerifySDK/TwilioVerify/Sources/TwilioVerifyError.swift | **`94.32%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**